sda-commons-shared-otel-agent/src/main/java/org/sdase/commons/shared/otel/agent/SdaConfigPropertyProvider.java

```diff
@@ -0,0 +1,38 @@
+package org.sdase.commons.shared.otel.agent;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.config.ConfigPropertySource;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This config source provider is used to define custom defaults that are relevant for sda services.
+ * A complete list of system properties that can be overridden can be found in <a
+ * href="https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md">Autoconfig
+ * module.</a>
+ */
+@AutoService(ConfigPropertySource.class)
+public final class SdaConfigPropertyProvider implements ConfigPropertySource {
+  private static final String JAEGER_SERVICE_NAME_ENV_VAR = "JAEGER_SERVICE_NAME";
+
+  @Override
+  public Map<String, String> getProperties() {
+    Map<String, String> properties = new HashMap<>();
+
+    // default otel collector gateway endpoint 4317 used for grpc
+    properties.put("otel.exporter.otlp.endpoint", "http://otel-collector-gateway.jaeger:4317");
```

Comment:
How does work for our customers or in different environments?

Reply:
This is just to define a default (still needs to be confirmed). This can be changed with the environment variable `OTEL_EXPORTER_OTLP_ENDPOINT` (documented in another module). We can add more documentation for deployment patterns... figured that this won't be the best place for  such docs.